imagebuildah/build.go

```diff
@@ -216,7 +216,7 @@ func BuildDockerfiles(ctx context.Context, store storage.Store, options define.B
 			Architecture: platform.Arch,
 			Variant:      platform.Variant,
 		})
-		if platformSpec.OS != "" && platformSpec.Architecture != "" {
+		if platformSpec.OS != "" || platformSpec.Architecture != "" {
```

Comment:
Break this into two separate blocks.  The first should overwrite `platformContext.OSChoice` if `platformSpec.OS` is set, the second should overwrite both `platformContext.ArchitectureChoice` and `platformContext.VariantChoice` if `platformSpec.Architecture` is set.

Reply:
done